FactoryMethodTask/Program.cs

```diff
@@ -1,13 +1,17 @@
 ﻿using System;
+using FactoryMethodTask.shapes;
+using FactoryMethodTask.shapes.bases;
 
-namespace FactoryMethodtask
+namespace FactoryMethodTask
 {
     public class Program
     {
         public static void Main(string[] args)
         {
-            IShape shape = ShapeFactory.GetRandomShape();
-
+            ClientShape client = new ClientShape();
+            
+            IShape shape = client.GetRandomShape();
```

Comment:
There should be only 1 access point. You shouldn't create IShape instance manually, it should be done in ClientShape class.
Your Main method should contain only 2 lines: creating a Shape and showing results in console. That's it!